### ISSUE TITLE
How to set a panel collapsed by default?

### ISSUE THREAD

**Issue by jsartisan** [bvaughn/react-resizable-panels#430]
Is there a way to collapse a panel by default?
I am creating this layout:
![CleanShot 2024-11-13 at 09 02 11](https://github.com/user-attachments/assets/db3d3944-9c0d-401e-bbff-1eab3e285012)
and I want the console panel to be collapsed by default. I don't see any prop that can do it.

**Comment by bvaughn** [comment]
Just set the default panel size (`defaultSize`) to the same thing as the collapsed size?

**Comment by jsartisan** [comment]
Tried this:

```jsx
const collapsedSize = (100 / parentSize) * 40;

  return (
    <ResizablePanel
      ref={ref}
      collapsible={true}
      minSize={10}
      collapsedSize={collapsedSize}
      defaultSize={props.collapsed ? collapsedSize : undefined}
      {...rest}
    >
      {content}
    </ResizablePanel>
  );
  ```
  
  but does not work. When running the above code, the resizing stops working for that group and the size defaults to 50%

**Comment by jsartisan** [comment]
Let me try again. Most likely i might be messing up some prop. I'll get back.

**Comment by jsartisan** [comment]
@bvaughn can we define collapsed size in pixels? i remember it was possible around 0.63 version.

**Comment by jsartisan** [comment]
Ok found a way to do it. By using flex-basis to whatever value i want and using collapsedSize and defaultSize as 0

```
<ResizablePanel
      ref={ref}
      collapsible={true}
      minSize={15}
      defaultSize={props.defaultCollapsed ? 0 : undefined}
      collapsedSize={0}
      className="!basis-10"
      {...rest}
    >
      {content}
    </ResizablePanel>
    ```
